pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go

```diff
@@ -121,17 +122,43 @@ func (c *ClusterRoleAggregationController) syncClusterRole(key string) error {
 		return nil
 	}
 
-	// we need to update
-	clusterRole := sharedClusterRole.DeepCopy()
-	clusterRole.Rules = nil
-	for _, rule := range newPolicyRules {
-		clusterRole.Rules = append(clusterRole.Rules, *rule.DeepCopy())
-	}
-	_, err = c.clusterRoleClient.ClusterRoles().Update(context.TODO(), clusterRole, metav1.UpdateOptions{})
+	clusterRole := rbacv1ac.ClusterRole(sharedClusterRole.Name).
+		SetRules(toApplyPolicyRules(newPolicyRules)...)
+
+	opts := metav1.ApplyOptions{FieldManager: "clusteraggregationcontroller", Force: true}
+	_, err = c.clusterRoleClient.ClusterRoles().Apply(context.TODO(), clusterRole, opts)
 
 	return err
 }
 
+func toApplyPolicyRules(rules []rbacv1.PolicyRule) []*rbacv1ac.PolicyRuleApplyConfiguration {
+	var result []*rbacv1ac.PolicyRuleApplyConfiguration
+	for _, rule := range rules {
+		result = append(result, toApplyPolicyRule(rule))
+	}
+	return result
+}
+
+func toApplyPolicyRule(rule rbacv1.PolicyRule) *rbacv1ac.PolicyRuleApplyConfiguration {
+	result := rbacv1ac.PolicyRule()
+	if rule.Resources != nil {
```

Comment:
Why does it matter that you're copying `nil`?

Reply:
Oh yeah, this is cruft from a previous revision of the code (where I had pointers to slices and so the test would fail because deepcopy can tell the difference between "nil" and "&nil"). Removing.